src/VDPRegViewer.cpp

```diff
@@ -35,20 +35,27 @@ void buttonHighlightDispatcher::receiveState(bool state)
 
 VDPRegViewer::VDPRegViewer(QWidget *parent)
 	: QDialog(parent)
+	, regs{0}
 {
 	setupUi(this);
 	connect(VDPcomboBox, qOverload<int>(&QComboBox::currentIndexChanged), this, &VDPRegViewer::on_VDPcomboBox_currentIndexChanged);
 
 	vdpId = 99; // make sure that we parse the first time the status registers are read
-	vdpId = VDP_V9958; //quick hack for now
+	vdpId = VDP_V9958; // quick hack for now
 
-	//now hook up some signals and slots
+	// now hook up some signals and slots
 	connectHighLights();
 
-	//get initial data
-	refresh();
+	// Now hook up some signals and slots.
+	// This allows the VDPDatastore to start asking for data as quickly as possible.
+	connect(&VDPDataStore::instance(), &VDPDataStore::dataRefreshed,
+		this, &VDPRegViewer::VDPDataStoreDataRefreshed);
```

Comment:
This is fine. Alternatively you could directly connect to `&VDPRegViewer::refresh`. Then you don't need the (trivial) `VDPRegViewer::VDPDataStoreDataRefreshed()` method (yet?).